### PR TITLE
Fix async generator-star issues

### LINKED ISSUE #3: Atom interprets `async` as error
Although the `eslint` bin does not complain, the `async` keyword is currently being interpreted as a violation of the `generator-star-spacing` rule on the [AtomLinter/linter-eslint plugin](https://github.com/AtomLinter/linter-eslint):

```js
async function foo() {
  await bar();
}
```

```
error  Unexpected space after *  generator-star-spacing
```

However, if I use the `eslint:recommended` extension and keep the `babel-eslint` as parser on `.eslintrc`, no error appears:

```yml
extends: 'eslint:recommended'
parser: 'babel-eslint'
```

Anyone else got this issue? I can always disable this rule on the `.eslintrc`, but since it works fine without this extension maybe it can be related with some configuration. 

Nice job with this module btw!

### PR DESCRIPTION
**Pull request by ruiquelhas**
Resolves #3.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -166,6 +166,7 @@ rules:
 
   # Babel plugin
   babel/arrow-parens: [2, as-needed]
+  babel/generator-star-spacing: 1
```
Comment (nunofgs) [inline .eslintrc]: I suggest `generator-star-spacing: [2, after]`. The only drawback is that both of these are allowed:

```js
function*()
function* ()
```

But that should be handled by another rule and should be added in a separate PR.

Comment (ruiquelhas) [inline .eslintrc]: :+1:
